File: dialogs/src/main/java/com/rands/lightbulb/dialogs/picker_dialog_adapter/AdapterPickerDialogSelection.java
package com.rands.lightbulb.dialogs.picker_dialog_adapter;

import com.rands.lightbulb.dialogs.base.BaseDialogSelection;

import java.util.Arrays;

class AdapterPickerDialogSelection extends BaseDialogSelection<int[]> {
    public AdapterPickerDialogSelection(int[] current, int[] draft) {
        super(current, draft);
    }

    @Override
    public boolean compareValues(int[] v1, int[] v2) {
        return Arrays.equals(v1, v2);
    }

    @Override
    public boolean hasCurrentSelection() {
        return getCurrentSelection() != null;
    }

    @Override
    public boolean hasDraftSelection() {
        return getDraftSelection() != null;
    }
}

File: lightbulb-dialogs/src/main/java/com/github/rooneyandshadowss/lightbulb/dialogs/picker_dialog_adapter/AdapterPickerDialogSelection.kt
package com.github.rooneyandshadowss.lightbulb.dialogs.picker_dialog_adapter

import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogSelection
import java.util.*

internal class AdapterPickerDialogSelection(current: IntArray?, draft: IntArray?) :
    BaseDialogSelection<IntArray?>(current, draft) {
    override fun compareValues(v1: IntArray?, v2: IntArray?): Boolean {
        return Arrays.equals(v1, v2)
    }

    override fun hasCurrentSelection(): Boolean {
        val currentSelection = getCurrentSelection()
        return currentSelection != null && currentSelection.isNotEmpty()
    }

    override fun hasDraftSelection(): Boolean {
        val draftSelection = getDraftSelection()
        return draftSelection != null && draftSelection.isNotEmpty()
    }
}